src/canvas.py

```diff
@@ -244,6 +243,19 @@ def remove_rubberband(self):
         self.canvas._rubberband_rect = None
         self.canvas.queue_draw()
 
+    # Overwritten function - do not change name
+    def push_current(self):
+        super().push_current()
```

Comment:
The difference with the default push_current is basically that it now automatically applies the limits again. Why is this necessary?